concent_api/core/tasks.py

```diff
@@ -29,12 +29,25 @@
 logger = logging.getLogger(__name__)
 
 
-@shared_task
+@shared_task(bind=True)
 @log_task_errors
 @transaction.atomic(using='control')
-def upload_finished(subtask_id: str):
+def upload_finished(self, subtask_id: str):
+    # Worker locks database row corresponding to the subtask in the subtask table.
     try:
-        subtask = Subtask.objects.get(subtask_id=subtask_id)
+        subtask = Subtask.objects.select_for_update(nowait=True).get(subtask_id=subtask_id)
```

Comment:
Why still `nowait`? Last time we talked about it, it turned out that `select_for_update()` can wait on its own and it's better to rely on it. So neither retry nor catching `DatabaseError` is necessary.

Reply:
right.. done